iqa_common/executor/execution_kubernetes.py

```diff
@@ -0,0 +1,226 @@
+import os
+import tempfile
+import threading
+import logging
+import urllib3
+from kubernetes.stream.ws_client import WSClient
+from kubernetes import config, client
+from kubernetes.client.apis import core_v1_api
+from kubernetes.stream import stream
+from iqa_common.executor import Execution, Command, ExecutionException
+
+
+# Logger for ExecutionKubernetes
+logger = logging.getLogger(__name__)
+urllib3.disable_warnings()
+
+
+class ExecutionKubernetes(Execution):
+    """
+    Represents the Execution of a command that is performed through the Kubernetes Client API (No local PID generated).
+    Executors that want to run a given command through the Kubernetes Client API must use this Execution strategy.
+    """
+
+    def __init__(self, command: Command, executor, modified_args: list=None, env=None):
+        """
+        Instance is initialized with the command that was effectively
+        executed and the Executor instance that produced this new object.
+        The KubernetesExecution expects that executor is an ExecutorKubernetes.
+        :param command:
+        :param executor:
+        :param modified_args:
+        :param env:
+        """
+        self.fh_stdout = None
+        self.fh_stderr = None
+
+        if command.stdout:
+            self.fh_stdout = tempfile.TemporaryFile(mode="w+t", encoding=command.encoding)
+        if command.stderr:
+            self.fh_stderr = tempfile.TemporaryFile(mode="w+t", encoding=command.encoding)
+
+        # Set the config and get Api instance
+        client_config = client.Configuration()
+        client_config.verify_ssl = False
+        client_config.assert_hostname = False
+        client_config.host = executor.host
+
+        # If a token has been provided use it
+        if executor.token:
+            client_config.api_key = {"authorization": "Bearer " + executor.token}
+
+        # Loading kubernetes config when config and context provided
+        if executor.config and executor.context:
+            config.load_kube_config(config_file=executor.config,
+                                    client_configuration=client_config,
+                                    context=executor.context)
+
+        # Kubernetes API instance
+        self.api = core_v1_api.CoreV1Api(client.ApiClient(client_config))
+
+        # Kubernetes response (internal execution)
+        self.response: WSClient = None
+
+        # Initializes the super class which will invoke the run method
+        super(ExecutionKubernetes, self).__init__(command=command, executor=executor,
+                                                  modified_args=modified_args, env=env)
+
+    def _run(self):
+        """
+        Run a separate thread to perform the command, so the thread can monitor
+        for command completion in a non-blocking way.
+        :return:
+        """
+        threading.Thread(target=self._run_as_thread).start()
+        while not self.response and not self.failure:
```

Comment:
How does this part work? I never user threads in python. 
While response & failure are not None, just carry on pass-looping? (so it is blocking wait?)